client/client.go

```diff
@@ -473,9 +478,15 @@ func filterDisabledRegions(regions []string, enabledRegions []types.Region) []st
 	}
 
 	var filteredRegions []string
-	for _, r := range regions {
-		if regionsMap[r] {
-			filteredRegions = append(filteredRegions, r)
+	if allRegions {
+		for region := range regionsMap {
+			filteredRegions = append(filteredRegions, region)
+		}
```

Comment:
This prevents there being regions not in our list, but that get returned by `ec2.DescribeRegions`

Reply:
Interesting edge-case, I'd add a comment in the code